CMakeLists.txt

```diff
@@ -41,6 +41,7 @@ option(MLX_ENABLE_X64_MAC "Enable building for x64 macOS" OFF)
 option(MLX_BUILD_GGUF "Include support for GGUF format" ON)
 option(MLX_BUILD_SAFETENSORS "Include support for safetensors format" ON)
 option(MLX_BUILD_BLAS_FROM_SOURCE "Build OpenBLAS from source code" OFF)
+option(MLX_BUILD_STUBS "Build stub files for python bindings" ON)
```

Comment:
I might rename that `MLX_BUILD_PYTHON_STUBS` just to make it a little more clear it's specific to Python.

Also we could just always build the stubs.. since it defaults to on seems unlikely anyone will ever turn it off.